docs/java/guides/cf-xsuaa.md

```diff
@@ -0,0 +1,229 @@
+
+1. Request:
+    ```
+    GET https://[xsuaa.url]/oauth/authorize
+    
+    Query parameters:
+    
+    client_id=[xsuaa.clientid]
+    redirect_uri=[application.route]
+    response_type=code
+    ```
+    (optional values can be set for "scope" and "login_hint")
+1. Submit login form via browser.
+2. Read HTTP response:
+    ```
+    HTTP/1.1 302 Found
+    Strict-Transport-Security: max-age=31536000
+    Set-Cookie: X-Uaa-Csrf=[...]; Path=/; Max-Age=86400; Expires=[...]; HttpOnly
+    Cache-Control: no-store
+    Content-Language: en
+    Location: [application.route]?code=[code]
+    X-XSS-Protection: 1; mode=block
+    X-Frame-Options: DENY
+    X-Content-Type-Options: nosniff
+    ```
+1. Extract `[code]` from `Location` header.
+
+### Get OAuth2 Access Token
+
+With the authorization code we can now request a real access token from the OAuth2 service endpoint:
+
+1. Request:
+    ```
+    POST https://[xsuaa.url]/oauth/token
+    
+    Headers
+    Accept: application/json
+    Content-Type: application/x-www-form-urlencoded
+   
+    client_id=[xsuaa.clientid]
+    client_secret=[xsuaa.clientsecret]
+    redirect_uri=[application.route]
+    code=[code]
+    grant_type=authorization_code
+    ```
+1. Response:
+    ```
+    {
+      "access_token": [access_token],
+      "token_type": "bearer",
+      "id_token": [...],
+      "refresh_token": [refresh_token],
+      "expires_in": [...],
+      "scope": [...],
+      "jti": [...]
+    }
+    ```
+   Congratulation, you have a valid `access_token`.
+   It can be further evaluated and forwarded.
+   
+   Some applications like _approuter_ will save the `refresh_token` to the user session.
+   This leverages the automatic retrieval of new access tokens, if the previous ones have expired during the active session.
+
+## JWT Bearer Token Grant
+
+Several services on SCP Cloud Foundry require a dedicated OAuth2 access token, e.g. _Connectivity Service_ and _Destination Service_.
+
+1. Open the **system-provided** environment variables of your application.
+
+1. In the JSON value, locate the object `VCAP_SERVICES.destination[0].credentials`.
+
+1. Read "clientid", "clientsecret", "uri".
+   
+1. Request:
+    ```
+    POST https://[xsuaa.url]/oauth/token
+    
+    Headers
+    Accept: application/json
+    Content-Type: application/x-www-form-urlencoded
+   
+    client_id=[destination.clientid]
+    client_secret=[destination.clientsecret]
+    assertion=[access_token]
+    grant_type=urn:ietf:params:oauth:grant-type:jwt-bearer
+    response_type=token
+    ```
+1. Response:
+    ```
+    {
+      "access_token": [destination_access_token],
+      "token_type": "bearer",
+      "expires_in": [...],
+      "scope": [...],
+      "jti": [...]
+    }
+   ```
+   Congratulation, you have a valid `destination_access_token`.
+   It can be used to query the `destination.uri` linked destination service on behalf of the current user.
```

Comment:
```suggestion
   Congratulation, you have a valid `destination_access_token`.
   It can be used to query the `destination.uri` linked to destination service on behalf of the current user.
```